### PR TITLE
`Exam mode`: Improve room distribution user interface

### PR DESCRIPTION
**Pull request by SamuelRoettgermann**
#### General
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [x] I **strictly** followed the [client coding guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] I **strictly** followed the [AET UI-UX guidelines](https://ls1intum.github.io/ui-ux-guidelines/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context
The UI for room distribution had a few design issues. I initially thought it would be a good idea to hide the Distribute button until at least one rooms is selected, which is a bad decision as it obfuscates where "the journey will go".
Then there was very little color coding to distinguish warnings from error messages and that was done in a confusing fashion (the `Distribute` button went from yellow/orange to a green).
This PR also addresses the rounding of the percentage that can in specific conditions show "Not enough seats [...] (100%)".

### Description
Updated the room distribution UI by changing the warning/success text to be an immediately identifiable color.
Made the room distribution button permanently present to not obfuscate where the distribution button will be.
Changed some translations to make the data more clear. 
Added a warning for no rooms selected.
Floor the percentage in the warning box instead of rounding it now.

### Steps for Testing
Prerequisites:
- TS3
- 1 Instructor
- 1 Exam with at least one registered student

1. Log in to Artemis
2. Navigate to Course Administration
3. Navigate to an exam
4. Go to Manage (i.e. Instructor actions)
5. Go to the "Students" tab
6. Click "Distribute"
7. Check that what is described in the `Description` section of this PR is present.

### Testserver States
You can manage test servers using [Helios](https://helios.aet.cit.tum.de/). Check environment statuses in the [environment list](https://helios.aet.cit.tum.de/repo/69562331/environment/list). To deploy to a test server, go to the [CI/CD](https://helios.aet.cit.tum.de/repo/69562331/ci-cd) page, find your PR or branch, and trigger the deployment.

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots
No rooms selected: 
image

Not enough rooms selected: 
image

Enough rooms selected:
image

## Summary by CodeRabbit

* **New Features**
  * Added warning notification when no room is selected in the student room distribution dialog.

* **Bug Fixes**
  * Adjusted percentage calculation for seat capacity display.

* **UI/UX Improvements**
  * Enhanced visual alerts for seat availability status messages.
  * Clarified room field label for better clarity.
  * Streamlined dialog action button interface.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/main/webapp/app/exam/manage/students/room-distribution/students-room-distribution-dialog.component.ts [hunk 1]
```diff
@@ -74,7 +74,7 @@ export class StudentsRoomDistributionDialogComponent {
         if (usableCapacity > totalStudents) {
             usableCapacity = totalStudents;
         }
-        const percentage: number = totalStudents > 0 ? Math.min(100, Math.round((usableCapacity / totalStudents) * 100)) : 0;
+        const percentage: number = totalStudents > 0 ? Math.min(100, Math.floor((usableCapacity / totalStudents) * 100)) : 0;
```
Comment (florian-glombik) [inline src/main/webapp/app/exam/manage/students/room-distribution/students-room-distribution-dialog.component.ts:82]: Might be good to add a test for the edge case that is fixed with this?

Comment (SamuelRoettgermann) [reply src/main/webapp/app/exam/manage/students/room-distribution/students-room-distribution-dialog.component.ts:82]: Yes, good idea

Comment (SamuelRoettgermann) [reply src/main/webapp/app/exam/manage/students/room-distribution/students-room-distribution-dialog.component.ts:82]: introduced it

File: src/main/webapp/app/exam/manage/students/room-distribution/students-room-distribution-dialog.component.spec.ts [hunk 2]
```diff
@@ -43,7 +45,7 @@ describe('StudentsRoomDistributionDialogComponent', () => {
     beforeEach(async () => {
         await TestBed.configureTestingModule({
             imports: [FaIconComponent, FormsModule],
-            declarations: [MockDirective(TranslateDirective), MockPipe(ArtemisTranslatePipe), MockComponent(HelpIconComponent)],
+            declarations: [MockDirective(TranslateDirective), MockPipe(ArtemisTranslatePipe, (key: string) => key), MockComponent(HelpIconComponent)],
```
Comment (florian-glombik) [inline src/main/webapp/app/exam/manage/students/room-distribution/students-room-distribution-dialog.component.spec.ts]: What does `(key: string) => key)` do?

File: src/main/webapp/app/exam/manage/students/room-distribution/students-room-distribution-dialog.component.ts [hunk 3]
```diff
@@ -84,18 +89,26 @@ export class StudentsRoomDistributionDialogComponent {
...
 
     closeDialog(): void {
         this.dialogVisible.set(false);
     }
 
     attemptDistributeAndCloseDialog(): void {
-        const selectedRoomIds = this.selectedRooms().map((room) => room.id);
+        const selectedRoomIds = this.selectedRooms()
+            .map((room) => room.id)
+            .filter((id) => id >= 0);
```
Comment (florian-glombik) [inline src/main/webapp/app/exam/manage/students/room-distribution/students-room-distribution-dialog.component.ts]: Can we get rid of the magic number here? Why does Id need to be greater than 0?

File: src/test/javascript/spec/helpers/mocks/service/mock-students-room-distribution.service.ts [hunk 4]
```diff
@@ -10,16 +10,9 @@ export class MockStudentsRoomDistributionService {
...
-    });
+    loadRoomData = jest.fn(() => {});
 
-    updateCapacityData = jest.fn((roomIds: number[], reserveFactor: number) => {
-        this.capacityData.set({
-            combinedDefaultCapacity: 0,
-            combinedMaximumCapacity: 0,
-        } as ExamDistributionCapacityDTO);
-    });
+    updateCapacityData = jest.fn((roomIds: number[], reserveFactor: number) => {});
```
Comment (florian-glombik) [inline src/test/javascript/spec/helpers/mocks/service/mock-students-room-distribution.service.ts:15]: Wasn't this a fine approach? What is the issue with it?

Comment (SamuelRoettgermann) [reply src/test/javascript/spec/helpers/mocks/service/mock-students-room-distribution.service.ts:15]: When I was mocking the functions in my test file it would overwrite my mocked implementation. I can't 100% remember the exact circumstances, but I remember thinking: "Man, this really isn't a bad approach, but it just does not work with my tests".

File: src/main/webapp/app/exam/manage/students/room-distribution/students-room-distribution-dialog.component.ts (hunk 3 above)
Comment (SamuelRoettgermann) [reply src/main/webapp/app/exam/manage/students/room-distribution/students-room-distribution-dialog.component.ts]: I'm really confused myself here. I thought I added some code that would automatically pre-select all rooms that are distributed to, and I assigned negative IDs to the custom, non-persisted rooms, but apparently I forgot to push some of the code or I reverted those changes because they would necessitate some more changes in the server-side, to properly handle custom positions, that does not necessarily need to be in this PR

File: src/main/webapp/app/exam/manage/students/room-distribution/students-room-distribution-dialog.component.spec.ts (hunk 2 above)
Comment (SamuelRoettgermann) [reply src/main/webapp/app/exam/manage/students/room-distribution/students-room-distribution-dialog.component.spec.ts]: It provides a (much simpler) mock for the mocked translate pipe. I can try and remove it again now, but for some reason it refused to work without it when I did this PR 2-3 weeks ago

Comment (SamuelRoettgermann) [reply src/main/webapp/app/exam/manage/students/room-distribution/students-room-distribution-dialog.component.spec.ts]: I managed to remove it without it breaking anything now

File: src/test/javascript/spec/helpers/mocks/service/mock-students-room-distribution.service.ts (hunk 4 above)
Comment (SamuelRoettgermann) [reply src/test/javascript/spec/helpers/mocks/service/mock-students-room-distribution.service.ts:15]: Cause it would always reset the mocked values I set

**Review by florian-glombik — APPROVED** [review]
Code

**Review by anian03 — APPROVED** [review]
Tested on TS5. I could verify that the distribute button was always visible (but disabled if not enough rooms were selected). The warnings about not having any rooms selected, and not having enough rooms make sense.

Not too important (because it makes no sense), but I noticed you can request "100% reserved seats" to be left empty, which means no distribution is possible. We could think about adding some warning that notes you can seat more students by lowering this percentage though
